src/test/java/core/basesyntax/service/impl/CsvDataReaderTest.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReaderService;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+class CsvDataReaderTest {
+
+    @Test
+    public void readFromFile_invalidPath_notOk() {
+        ReaderService readerService = new CsvDataReader();
```

Comment:
Service should be a class field, 
try to use one instance of service, strategy, operation, repository
fix it everywhere